Installing-and-Configuring/RecoveryShell.md

```diff
@@ -0,0 +1,94 @@
+
+The recovery shell wipes secrets--normally used for security checks--that were [computed](/Keys/#tpm-pcrs) from the BIOS, kernel modules loaded, etc. This will limit the features available in this environment.
+
+
+Boot Process
+----
+
+To troubleshoot you should understand the processes used to configure and boot heads after flashing it into the BIOS.
+
+### configuration
+
+* First Boot Checks
+* set default boot and flash BIOS (again)
+* Reset the TPM
+* sign files in /boot
+* reset htop/totp
+
+### boot
+
+* mount default boot
+* check signatures
+* check hotp
+* menus and user interaction
+* boot kernel
+
+
+Troubleshoot the boot process
+----
+
+### manual boot
+
+If you want or need to manually boot a Linux system you must specify a kernel, initrd, and root file system.  Use the kexec-boot utility.  In this example 'foo' is a description that normally comes from other parts of heads configurations.  It can be anything.  The root filesystem must be the correct one used in the target Linux installation.  
+
+This example may work for you by changing only the root= setting.  Normally, there are symlinks in the boot partition using these file names.  Of course you need to adjust if your target system uses a different convention for specifying these files.
+
+    kexec-boot -b /boot -e ‘foo|elf|kernel /vmlinuz|initrd /initrd.img|append root=/dev/whatever’
+
+
+### sign files
+
+Content in /boot is hashed and recorded in a file.  The hashes are signed using the security dongle paired with heads.  These hashes are verified on boot using the public key corresponding to the security dongle.
+
+    mount /dev/sdaX /boot
+    kexec-sign-config -p /boot
+
+Heads uses a gpg detached signed digest for this process.  First, it finds all filenames in /boot which start with kexec and pass this as arguments to sha256sum. The output is redirected to gpg which verifies that the hashes for each file are the same as expected. Using sha256sum is faster than asking gpg to verify the files directly.  Creating the detached signed digest requires the user to sign externally with his USB Security dongle + User PIN, while the verification only requires the GPG public key fused inside of the rom.
+
+/boot/kexec_hashes.txt is a sha256sum (digest) file of all /boot content.  /boot/kexec.sig is a detached signed digest (sha256sum) of all kexec* files, including the kexec_hashes.txt file.  The function verify_global_hashes (which is duplicated in both gui-init and kexec-select-boot) verifies that the copied kexec_hashes.txt over /tmp matches the actual calculated hashes of /boot/ content prior of booting.
+
+On boot, kexec.sign is verifying detached signed integrity/authenticity of that signed file against the public key fused in rom, which was detached signed with the USB Security dongle's private key, protected in the USB Security dongle's smartcard, unlocked to sign with GPG User PIN.
+
+### hotp and totp
+
+Will not work in recovery shell due to missing secrets. 
```

Comment:
partly covered [here](https://github.com/osresearch/heads-wiki/pull/71/files#diff-14da7bd28be8213a9f230ea3c67a3fa86ba7e24c5ebc53d992d06e6453d9e1bdR34) which is part duplicate. maybe we want to link sections together?

Reply:
@tlaurion changed to reference Limitations section above.  Will push in a minute.